library/std/src/env.rs

```diff
@@ -198,13 +198,16 @@ impl fmt::Debug for VarsOs {
 ///
 /// # Errors
 ///
-/// This function will return an error if the environment variable isn't set.
+/// This function returns [`VarError::NotPresent`] if the environment variable
+/// isn't set.
 ///
-/// This function may return an error if the environment variable's name contains
-/// the equal sign character (`=`) or the NUL character.
+/// This function may return [`VarError::NotPresent`] if the
+/// environment variable's name contains the equal sign character (`=`) or the
+/// NUL character.
```

Comment:
can we frame this in a way that visibly ORs these two together? like

```suggestion
/// This function may return [`VarError::NotPresent`] if
/// - the environment variable isn't set
/// - the environment variable's name contains an equals sign or NUL (`=` or `\0`)
```

Reply:
I suppose we could leave it as you have written it since it's copying directly from `var_os` for the most part, but it feels strange to me that they are being made to match here but one has an `# Errors` header?

![errors header creating this additional visual distinction](https://github.com/user-attachments/assets/f81c21c5-7128-45fc-8e0a-17936457ce98)

There doesn't seem to be, even in this module, consistency in whether that is present on Result-returning functions, or what it documents:
- https://doc.rust-lang.org/std/env/fn.set_current_dir.html
- https://doc.rust-lang.org/std/env/fn.current_exe.html
- https://doc.rust-lang.org/std/env/fn.current_dir.html

It doesn't seem to make things clearer the way the `# Examples` header tends to.

Reply:
The main reason I bring this up is that for the most part they are self-explanatory variants except the "returns `NotPresent` if we determine the variable is invalid" happens yes, but is an essentially spurious error, and `Option` requires this because it doesn't attach a direct semantic to the `Some` or `None` variants, being a much more "vibe-based" notion of "was available".

Reply:
Excellent suggestions, thanks! I have pushed a new revision:

* Uses a bullet list for the error results. This is much clearer, thanks! This makes it look similar to [current_dir](https://doc.rust-lang.org/std/env/fn.current_dir.html).
* Minor tweaks to make the text a bit shorter.

I have left the Errors header, but if you want it removed, I'm happy to do that also. I *think* the general recommendation is that functions returning a `Result` should document the errors in an `Errors` section. Some evidence:

* The [API guide recommends it ](https://rust-lang.github.io/api-guidelines/documentation.html#function-docs-include-error-panic-and-safety-considerations-c-failure): "Error conditions should be documented in an "Errors" section"
* Clippy's pedantic lint that warns about functions using a Result without an Errors section: https://rust-lang.github.io/rust-clippy/master/#/missing_errors_doc
* In the `std::env` module, `current_exe`, `current_dir`, `join_paths`, and `var` currently all include an Errors section. The only function that returns a Result without an Errors section is `set_current_dir`. I can submit a separate change to add an Errors section for that function as well.

Current rendered version:


![errors](https://github.com/user-attachments/assets/69bd5c82-5445-4af1-9b23-52746e4f95ae)